### PR TITLE
Inhibit upgrade if multiple kernel-devel pkgs are installed

### PR DESCRIPTION
**Pull request by drehak**
When multiple `kernel-devel` packages are present, leapp attempts to upgrade them, but can't resolve the transaction in most cases. So we've decided to inhibit the upgrade and present the user with hints on removing the surplus devel kernels.

There's also a high chance of unexpected issues occuring on older kernels, so we've decided to inhibit if the machine isn't booted into the latest installed one.

### CODE REVIEW

**Comment by examon** [comment]
leapp-ci build

**Comment by drehak** [comment]
leapp-ci build

File: repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rpms = next(api.consume(InstalledRedHatSignedRPM), InstalledRedHatSignedRPM())
+    return [pkg for pkg in rpms.items if pkg.name == 'kernel-devel'].sort(key=lambda pkg: pkg.version)
+
+
+def process():
+    pkgs = get_kernel_devel_rpms()
+    if len(pkgs) > 1:
+        title = 'Multiple devel kernels installed'
+        summary = ('DNF cannot produce a valid package transaction when multiple instances of package'
+                   ' kernel-devel are installed.')
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py]:
```suggestion
        summary = ('DNF cannot produce a valid upgrade transaction when multiple'
                   ' kernel-devel packages are installed.')
```

File: repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM
+
+
+def get_kernel_devel_rpms():
+    rpms = next(api.consume(InstalledRedHatSignedRPM), InstalledRedHatSignedRPM())
+    return [pkg for pkg in rpms.items if pkg.name == 'kernel-devel'].sort(key=lambda pkg: pkg.version)
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py]: versions are same. you should focus on release

Comment (drehak) [reply repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py]: Ah, thanks. I confused the fields. :|

File: repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py (hunk 1 above)
Comment (drehak) [reply repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/library.py]: Got it, thanks.

**Comment by pirat89** [comment]
@drehak  see pls #464

File: repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/actor.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from leapp.libraries.actor import library
+
+
+class CheckInstalledDevelKernels(Actor):
+    """
+    Inhibit IPU (in-place upgrade) when multiple devel kernels are installed.
+
+    Because of an issue in DNF, the transaction can't be validated if there's
+    more than one package named kernel-devel. Therefore, in this case, we
+    inhibit the ugprade with a clearer remediation.
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/actor.py]:
```suggestion
    inhibit the upgrade with a clearer remediation.
```

**Comment by drehak** [comment]
Currently there's no inhibition with two kernel-devel packages, I'll look into that.

**Comment by drehak** [comment]
e2e tests

**Comment by drehak** [comment]
leapp-ci build

**Comment by bocekm** [comment]
@drehak, please squash.

**Comment by drehak** [comment]
@bocekm my bad, I've squashed but forgot to push.

**Comment by pirat89** [comment]
Lgtm and tested.

**Pull request merged**